client/src/Utxorpc/Client.hs

```diff
@@ -0,0 +1,154 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RankNTypes #-}
+
+module Utxorpc.Client
+  ( utxorpcService,
+    fromConfig,
+    withHeaders,
+  )
+where
+
+import qualified Data.ByteString as BS
+import Network.GRPC.Client (gzip, uncompressed)
+import Network.GRPC.Client.Helpers
+  ( GrpcClient (_grpcClientHeaders),
+    GrpcClientConfig,
+    UseTlsOrNot,
+    close,
+    grpcClientConfigSimple,
+    setupGrpcClient,
+    _grpcClientConfigCompression,
+  )
+import Network.GRPC.HTTP2.ProtoLens (RPC (RPC))
+import Network.HTTP2.Client (ClientIO, HostName, PortNumber)
+import Proto.Utxorpc.Build.V1.Build
+import Proto.Utxorpc.Submit.V1.Submit
+import Proto.Utxorpc.Sync.V1.Sync
+import Proto.Utxorpc.Watch.V1.Watch
+import Utxorpc.Logged (UtxorpcClientLogger, loggedSStream, loggedUnary)
+import Utxorpc.Types
+
+utxorpcService ::
```

Comment:
For public library code you should really be including Haddock comments for each exported item (type, function, etc.)

I expect some of your existing comments can be repurposed, just by adding the `|` markup.

Note that you can make module comments, too. However, have a look at the generated docs to make sure things make sense in the order they come out.

Reply:
I will work on this in a future PR.